src/main/java/crimson/CommitStatusHandler.java

```diff
@@ -0,0 +1,60 @@
+package crimson;
+import io.github.cdimascio.dotenv.Dotenv;
+import kotlin.Pair;
+import org.apache.http.HttpEntity;
+import org.apache.http.HttpHeaders;
+import org.apache.http.HttpResponse;
+import org.apache.http.client.methods.HttpPost;
+import org.apache.http.entity.StringEntity;
+import org.apache.http.impl.client.HttpClients;
+import org.apache.http.client.HttpClient;
+import java.io.IOException;
+import java.util.*;
+
+public class CommitStatusHandler {
+    static Dotenv dotenv = Dotenv.load();
+
+    public static final String ERROR = "error";
+    public static final String FAILURE = "failure";
+    public static final String PENDING = "pending";
+    public static final String SUCCESS = "success";
+
+    private static final String USER_NAME = "641bill";
+    static final String GITHUB_TOKEN = dotenv.get("GITHUB_TOKEN");
+    private static final String API_URL = "https://api.github.com/repos/kth-dd2480-vt22/assignment2/statuses/";
+
+    private static final String context = "Group13-CI";
+
+    /**
+     * Set the commit status on GitHub to be the given state along with other arguments,
+     * using GitHub REST API. It is done by sending a HTTP request to the GitHub API url.
+     * @param state
+     * @param target_url
+     * @param description
+     * @param commitHash
+     * @return the HttpEntity of the response
+     * @throws IOException
+     */
+    public HttpEntity setStatus(String state, String target_url, String description, String commitHash) throws IOException {
```

Comment:
I don't think we need to return this too.

Reply:
This is just for testing

Reply:
Do I need to delete all the tests also because it's not gonna work without a .env

Reply:
You can rewrite the test so that it read from some file (or don't make the call in the test)

Testing with third parties API is always tricky in API test

Reply:
I would suggest we don't test writing to GitHub as it's a issue to commit token into code.

Reply:
True, and not testing it makes sense so I just deleted it